functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -0,0 +1,278 @@
+			
+			if ($server.VersionMinor -eq 50) {
+				$version = "2008R2"
+			}
+			else {
+				$version = switch ($server.VersionMajor) {
+					9 { "2005" }
+					10 { "2008" }
+					11 { "2012" }
+					12 { "2014" }
+					13 { "2016" }
+					14 { "2017" }
+				}
+			}
+			
+			if (!$instanceOnly) {
+				$databases = Invoke-Sqlcmd2 -ServerInstance $server -Database master -Query "Select Name from sys.databases where name not in ('master', 'model', 'msdb', 'tempdb')"
+			}
+			
+			$script = $scriptversions | Where-Object -Property Version -eq $version | Select-Object -ExpandProperty Script
+			
+			if ($null -eq $first) { $first = $true }
+			if ($useSelectionHelper -and $first) {
+				$queryName = Invoke-DiagnosticQuerySelectionHelper $script
+				$first = $false
+			}
+			
+			if (!$instanceonly -and !$databaseSpecific -and !$queryName) {
+				$scriptcount = $script.count
+			}
+			elseif ($instanceOnly) {
+				$scriptcount = ($script | Where-Object DatabaseSpecific -eq $false).count
+			}
+			elseif ($databaseSpecific) {
+				$scriptcount = ($script | Where-Object DatabaseSpecific).count
+			}
+			elseif ($queryName.Count -ne 0) {
+				$scriptcount = $queryName.Count
+			}
+			
+			foreach ($scriptpart in $script) {
+				
+				if (($queryName.Count -ne 0) -and ($queryName -notcontains $scriptpart.QueryName)) { continue }
+				if (!$scriptpart.DatabaseSpecific -and !$databaseSpecific) {
+					if ($PSCmdlet.ShouldProcess($instance, $scriptpart.QueryName)) {
+						$counter++
+						if (!$silent) {
+							Write-Progress -Id 1 -ParentId 0 -Activity "Collecting diagnostic query data from $instance" -Status "Processing $counter of $scriptcount" -CurrentOperation $scriptpart.QueryName -PercentComplete (($counter / $scriptcount) * 100)
+						}
+						
+						
+						try {
+							$result = Invoke-Sqlcmd2 -ServerInstance $server -Database master -Query $scriptpart.Text -ErrorAction Stop
+							Write-Message -Level Output -Message "Processed $($scriptpart.QueryName) on $instance"
+							
+							if (!$result) {
+								$result = [pscustomobject]@{
+									ComputerName = $server.NetName
+									InstanceName = $server.ServiceName
+									SqlInstance = $server.DomainInstanceName
+									Number = $scriptpart.QueryNr
+									Name = $scriptpart.QueryName
+									Description = $scriptpart.Description
+									DatabaseSpecific = $scriptpart.DBSpecific
+									DatabaseName = $null
+									Notes = "Empty Result for this Query"
+									Result = $null
+								}
+								Write-Message -Level Verbose -Message ("Empty result for Query {0} - {1} - {2}" -f $scriptpart.QueryNr, $scriptpart.QueryName, $scriptpart.Description)
+							}
+						}
+						catch {
+							Write-Message -Level Verbose -Message ('Some error has occured on Server: {0} - Script: {1}, result unavailable' -f $instance, $scriptpart.QueryName)
```

Comment:
Consider adding the `-Target` parameter, either containing the script or the instance
Consider adding the `-ErrorRecord $_` parameter, to help with debugging and ensure expected unattended behavior